libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1190,23 +1198,28 @@ void IRGeneratorForStatements::appendExternalFunctionCall(
 		encodeInPlace = false;
 	bool encodeForLibraryCall = funKind == FunctionType::Kind::DelegateCall;
 	solUnimplementedAssert(!encodeInPlace, "");
-	solUnimplementedAssert(!funType.padArguments(), "");
+	solUnimplementedAssert(funType.padArguments(), "");
```

Comment:
This looks a bit suspicious... This function is called for bare calls, which will have this set to ``false`` - apparently there are no tests for that? We should have a quick look into this.